src/nrnoc/md1redef.h

```diff
@@ -21,7 +21,6 @@
 #undef nodecount
 #undef type
 
-#define NrnThread _NrnThread
```

Comment:
>  using _NrnThread triggers undefined behaviour.

So ```#define NrnThread _aNrnThread``` would suffice along with equivalent everywhere else. However, as this is a case with very unlikely user name conflict, the whole nocmodl and mod2c is on the road to replacement by NMODL, and an ```_a``` prefix adds a new example of obscurity.  I accept the change to ```NrnThread``

Reply:
OK! Just to note that [identifiers that begin with an underscore are reserved in the global namespace](https://en.cppreference.com/w/cpp/language/identifiers) would also hit the suggestion of `_aNrnThread`. In some future PR it might make sense to introduce a `neuron` namespace to insulate NEURON a bit better from the rest of the world.